core/src/main/scala/org/apache/spark/deploy/history/FsHistoryProvider.scala

```diff
@@ -541,6 +542,23 @@ private[history] class FsHistoryProvider(conf: SparkConf, clock: Clock)
     }
   }
 
+  private[history] def checkAbsoluteLength(info: LogInfo, entry: FileStatus): Boolean = {
```

Comment:
The name of the method and the return value are a little cryptic. What does it mean to check?

Might be better to calls it something like `shouldReloadLog` or something. You could also move the existing check into this function and make the call site simpler.